apps/frontend/src/app/2_molecules/BalanceInput/BalanceInput.tsx

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { forwardRef, useImperativeHandle, useRef } from 'react';
+
+import { BigNumberish } from 'ethers';
+
+import { SupportedTokens } from '@sovryn/contracts';
+import { ChainId } from '@sovryn/ethers-provider';
+import { AmountInput, AmountInputProps } from '@sovryn/ui';
+
+import { useMaxAssetBalance } from '../../../hooks/useMaxAssetBalance';
+import { getRskChainId } from '../../../utils/chain';
+import { fromWei } from '../../../utils/math';
+
+const DEFAULT_GAS_LIMIT = 1000000; // 1M wei
```

Comment:
I'd change it to `6 800 000`, see the comment from @john-light : https://sovryn.atlassian.net/browse/SOV-1013?focusedCommentId=11653 . I'd also rename it to `MAX_GAS_LIMIT` and move it somewhere else as it's used in `useMaxAssetBalance`.

@john-light are you ok with us subtracting `gasLimit * gasPrice` if we have a custom gas limit passed to the component and use `DEFAULT_GAS_LIMIT` only as a fallback or do you want us to use it all the time?

Reply:
Having it as a fee in the max amount makes sense and makes it easer for us to handle.

In case there was a plan to use this fee in the actual transaction - noting that 6.8M is overkill to use and should be avoided whenever possible.

cc @john-light

Reply:
> are you ok with us subtracting gasLimit * gasPrice if we have a custom gas limit passed to the component and use DEFAULT_GAS_LIMIT only as a fallback or do you want us to use it all the time?

@tiltom 
Can you suggest a scenario where the user is passing a custom gas limit _before_ selecting the amount of their tx?

If you cannot suggest such a scenario, then for the purposes of calculating "max amount" I request that you use the formula I specified in Jira.

> In case there was a plan to use this fee in the actual transaction - noting that 6.8M is overkill to use and should be avoided whenever possible.

@creed-victor 
Yes, unless the tx the user is making really requires 6.8M gas, a more reasonable default gas limit should be used. (There is a different story for defining default gas limits [here](https://sovryn.atlassian.net/browse/SOV-1452).)

Reply:
Ok, this conversation can be resolved then